### PR TITLE
Move the HTML block into the blocks library package

### PR DESCRIPTION
**Pull request by youknowriad**
In order to merge Gutenberg into Core easily, all the Gutenberg JavaScript Core must be shared a reusable npm package. This PR updates the HTML block to avoid relying on `wp.codeEditor` for now because there's no easy way to lazy-load WordPress scrips yet and move the `HTML` block into the blocks library.

The `wp.components.CodeEditor` has also been deprecated with this change.

**Testing instructions**

 - Make sure you can use HTML block properly.
 - There's no syntax highlighting for the moment but it's consistent with the Code Block and could be added later in a better way.

### CODE REVIEW

File: components/code-editor/index.js [hunk 1]
```diff
@@ -76,6 +77,11 @@ class LazyCodeEditor extends Component {
 		this.state = {
 			status: 'pending',
 		};
+
+		deprecated( 'wp.components.CodeEditor', {
+			version: '4.2',
+			alternative: 'wp.codeEditor directly or any alternative React component for syntax highlighting',
```
Comment (gziolo) [inline components/code-editor/index.js]: This is going to be confusing in the context of npm package which is out of WordPress context. Not sure if this is bad or not, just noting.

**Review by gziolo — APPROVED** [review]
It looks good from my perspective.

Should we get design feedback for this one as it changes the UX for the Code block?

Should we merge #10397 in this PR and test them together again? I think we might want to add Code block inside the changelog, the way we did it for Classic block in #10397.

File: components/code-editor/index.js (hunk 1 above)
Comment (youknowriad) [reply components/code-editor/index.js]: yes, we have a bunch of those and we need to experiment with making the messages opt-in or something and only enable in WP

**Comment by chrisvanpatten** [comment]
Sad to lose syntax highlighting but it looks good to me, especially since it mirrors the code block. Hopefully in Phase 2 it'll be possible to get a better code editing solution in place across the board. 

Only thing to wonder about is whether custom HTML and code will get confused for each other but honestly I think we have that problem either way.

👍  from me!

**Comment by ZebulanStanphill** [comment]
I'm worried that people are going to complain a lot about the loss of functionality. It is easy to explain to a developer why the syntax highlighting may have to be removed for now, but it is far harder to explain to the end user. To some, I am sure this will seem like a sign that Gutenberg is cutting corners to try and meet the deadline.

Personally, I REALLY like the syntax highlighting and would not like to see it be removed at all. If anything, I wish the standard <kbd><samp>Edit as HTML</samp></kbd> option and Code Editor mode used syntax highlighting.

**Comment by youknowriad** [comment]
Agreed, it would be great to have syntax Highlighting but personally, I do think the current syntax highlighting is not great anyway and should be improved. We should explore alternatives to CodeMirror. (Monaco Editor is a very good option these days).

Removing this from the npm package (where shipping Lazy Loading in a generic way is not really possible) doesn't mean we can't intergrate it back from Core Side but I do think we gain clarity by doing so as the HTML block is something any Gutenberg install (CMS agnostic) could benefit from.

**Comment by jasmussen** [comment]
Thanks for working on this.

This is an interesting crossroads to face. The syntax highlighter is delicious and wonderful, but at the same time if we can't lazy-load it now, it comes with a performance hit. 

Difficult. But in order to not stall momentum, I think we should do this:

- Create a future-milestoned ticket to add back the code editor when we can lazy load it
- Ship this in the meantime. 

> To some, I am sure this will seem like a sign that Gutenberg is cutting corners to try and meet the deadline.

This is the difficulty of making software. On one hand there's a great desire to have "everything ready" for the first version. On the other hand, the longer we wait to ship, the longer we deny ourselves the much needed testing surface that is the entire web. Although Gutenberg at the time of writing is on [530.000 sites](https://gutenstats.blog/), there are still many rough corners we need to sand off before shipping which means we need to make tough decisions related to focus. In that vein, the most important thing is to [ship a 1.0](https://ma.tt/2010/11/one-point-oh/).

On the flipside, if we all pull together and ship the 1.0, then the [1.1 can hopefully be released as early as March 2019](https://make.wordpress.org/core/2018/10/03/proposed-wordpress-5-0-scope-and-schedule/).

**Comment by youknowriad** [comment]
I created this issue to track it #10423

**Pull request merged**